tests/estimator/ops/test_estimator_symbolics.py

```diff
@@ -57,23 +56,15 @@ def test_resource_decomp(self):
             **op.resource_params
         )
 
-        class ResourceDummyS(qre.S):
-            """Dummy class with no default adjoint decomp"""
-
-            @classmethod
-            def adjoint_resource_decomp(cls, target_resource_params=None) -> list[GateCount]:
-                """No default resources"""
-                raise ResourcesUndefinedError
-
-        op = ResourceDummyS()  # no default_adjoint_decomp defined
```

Comment:
These `ResourceDummy` classes are being removed because they no longer serve the purpose of testing the path where an Op does not have an `adjoint_resource_decomp` defined and `resource_decomp` is called on its Adjoint. For example, this `ResourceDummy` class, inherits `qre.S`'s `adjoint_resource_decomp` method rather than `ResourceOperator`'s `adjoint_resource_decomp` method. Thus, instead of using `ResourceDummy`, we just use `QubitUnitary`, which allows us to test that the `ResourceOperator`'s `adjoint_resource_decomp` is indeed called when the Op does not define an `adjoint_resource_decomp`.